### PR TITLE
Refactor OpenAI Chat converter into template-style design

### PR DESCRIPTION
**Pull request by nv-hwoo**
This PR
- simplifies and refactors `OpenAIChatCompletionsConverter` into template-style design
- create unit tests for the converter class that handles chat, vision, and image retrieval endpoints.

This PR does _NOT_
- change any code behavior (except that it assumes there's no dataset approach anymore, which is getting handled by #100 )

### CODE REVIEW

File: genai-perf/tests/test_input_image_retrieval.py [hunk 1]
```diff
@@ -1,155 +0,0 @@
-# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (debermudez) [inline genai-perf/tests/test_input_image_retrieval.py:1]: why did this whole file get deleted?

Comment (nv-hwoo) [reply genai-perf/tests/test_input_image_retrieval.py:1]: From the PR description above,
> create unit tests for the converter class that handles chat, vision, and image retrieval endpoints.

Basically ported to/under `test_openai_chat_converter.py`

Comment (debermudez) [reply genai-perf/tests/test_input_image_retrieval.py:1]: oof. shouldn't have missed that.
thank you

File: genai-perf/tests/test_output_format_converter.py [hunk 2]
```diff
@@ -23,6 +24,52 @@
 
 
 class TestOutputFormatConverter:
+
```
Comment (debermudez) [inline genai-perf/tests/test_output_format_converter.py:27]: nice test to ensure we don't break this.

**Review by debermudez — COMMENTED** [review]
approval after the code ql fix

**Review by debermudez — APPROVED** [review]

**Pull request merged**